scripts/generate-cluster-creds.sh

```diff
@@ -89,22 +89,19 @@ EOF
     rm kubeconfig.yaml
   fi
 
-  yaml_creds=$(yq '.k8s_config' /tmp/temp_k8sconfig.yaml)
-  json_creds=$(yq e -o json '.k8s_config' /tmp/temp_k8sconfig.yaml | jq -r -M -c .)
-
+  output=$(yq '.k8s_config' /tmp/temp_k8sconfig.yaml)
   # NOTE: Write to ci/ files as these files will be used in ci tests! Consider looping through all files
   # in ci dir
-  for APP in merlin turing
+  for CHART in $(echo $CHARTS_CHANGED | tr ' ' '\n' | grep 'charts/\(merlin\|turing\|caraml\)' | sed -r 's/charts\///')
```

Comment:
Over here we iterate over a filtered list of charts (`merlin`/`turing`/`caraml` only but not necessarily all of them) and apply the appropriate changes to the CI values file. For the `caraml` chart we need to inject the credentials for both the `merlin` and `turing` subchart, hence the nested loop.